jetcd-core/src/main/java/com/coreos/jetcd/Watch.java

```diff
@@ -48,11 +48,12 @@
   /**
    * Interface of Watcher.
    */
-  interface Watcher {
+  interface Watcher extends AutoCloseable {
 
     /**
      * closes this watcher and all its resources.
      **/
+    @Override
     void close();
```

Comment:
AutoCloseable::close() throws an exception, I do not think we should change the signature

Reply:
I think the point is whether to throw out exception on `Watch#close`, so far it’s prohibited.

We uses `AutoCloseable` just for `try-with-resource`, it should not be a constraint.

Reply:
so lgtm :)